platform/posix/transport/src/mbedtls_pkcs11_posix.c

```diff
@@ -0,0 +1,958 @@
+ * of this transport implementation is included to your project.
+ *
+ * @note When using multiple transports in the same compilation unit,
+ *       define this pointer as void *.
+ */
+struct NetworkContext
+{
+    MbedtlsPkcs11Context_t * pParams;
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Represents string to be logged when mbedTLS returned error
+ * does not contain a high-level code.
+ */
+static const char * pNoHighLevelMbedTlsCodeStr = "<No-High-Level-Code>";
+
+/**
+ * @brief Represents string to be logged when mbedTLS returned error
+ * does not contain a low-level code.
+ */
+static const char * pNoLowLevelMbedTlsCodeStr = "<No-Low-Level-Code>";
+
+/**
+ * @brief Utility for converting the high-level code in an mbedTLS error to string,
+ * if the code-contains a high-level code; otherwise, using a default string.
+ */
+#define mbedtlsHighLevelCodeOrDefault( mbedTlsCode )       \
+    ( mbedtls_high_level_strerr( mbedTlsCode ) != NULL ) ? \
+    mbedtls_high_level_strerr( mbedTlsCode ) : pNoHighLevelMbedTlsCodeStr
+
+/**
+ * @brief Utility for converting the level-level code in an mbedTLS error to string,
+ * if the code-contains a level-level code; otherwise, using a default string.
+ */
+#define mbedtlsLowLevelCodeOrDefault( mbedTlsCode )       \
+    ( mbedtls_low_level_strerr( mbedTlsCode ) != NULL ) ? \
+    mbedtls_low_level_strerr( mbedTlsCode ) : pNoLowLevelMbedTlsCodeStr
+
+/**
+ * @brief Debug logging level to use for MbedTLS.
+ */
+#define MBEDTLS_DEBUG_LEVEL    0
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Initialize the MbedTLS structures in a network connection.
+ *
+ * @param[in] pContext The SSL context to initialize.
+ */
+static void contextInit( MbedtlsPkcs11Context_t * pContext );
+
+/**
+ * @brief Free the MbedTLS structures in a network connection.
+ *
+ * @param[in] pContext The SSL context to free.
+ */
+static void contextFree( MbedtlsPkcs11Context_t * pContext );
+
+/**
+ * @brief Set up TLS on a TCP connection.
+ *
+ * @param[in] pMbedtlsPkcs11Context Network context.
+ * @param[in] pHostName Remote host name, used for server name indication.
+ * @param[in] pMbedtlsPkcs11Credentials TLS setup parameters.
+ * @param[in] recvTimeoutMs Receive timeout for network socket.
+ *
+ * @return #MBEDTLS_PKCS11_SUCCESS, #MBEDTLS_PKCS11_INSUFFICIENT_MEMORY, #MBEDTLS_PKCS11_INVALID_CREDENTIALS,
+ * #MBEDTLS_PKCS11_HANDSHAKE_FAILED, or #MBEDTLS_PKCS11_INTERNAL_ERROR.
+ */
+static MbedtlsPkcs11Status_t configureMbedtls( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                               const char * pHostName,
+                                               const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials,
+                                               uint32_t recvTimeoutMs );
+
+/**
+ * @brief Configure the credentials in the MbedTLS SSL context.
```

Comment:
Please mention which types certificates (Client, Root CA) are configured by the function